pkg/test/environment.go

```diff
@@ -135,6 +137,8 @@ type Environment struct {
 	gatewayConfig *v1beta1.GatewayConfig
 	k8sEnv        *envtest.Environment
 
+	embeddedJS *natsserver.Server
```

Comment:
Could the lifetime of this server be controlled by ctx directly?

Reply:
There doesn't seem to be any wrappers that take a context so we would need to do this ourselves if we wanted to.

Reply:
In that case we should do it ourselves by adding a waitctx action.